MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -341,10 +341,9 @@
 
     VkExtent2D surfExtnt = mvkVkExtent2DFromCGSize(mtlLayer.naturalDrawableSizeMVK);
 
-	// Metal supports 3 concurrent drawables, but if the swapchain is destroyed and
-	// rebuilt as part of resizing, one will be held by the current display image.
-	pSurfaceCapabilities->minImageCount = 2;
-	pSurfaceCapabilities->maxImageCount = 2;
+	// Metal supports 3 concurrent drawables.
+	pSurfaceCapabilities->minImageCount = 3;
```

Comment:
This would indicate to apps that they can't request double buffering.

Reply:
@cdavis5e Thanks for the sanity check.

Hmmm...yeah...I pondered that when I changed it.

But then I wondered how an app _would_ request double-buffering. From how I read the spec, `VkSwapchainCreateInfoKHR ::minImageCount` is just that...a _minimum_...which the implementation may be free to exceed if it works better for the implementation:

> `minImageCount` is the minimum number of presentable images that the application needs. The implementation will either create the swapchain with at least that many images, or it will fail to create the swapchain.

If the app passed 2 in for `minImageCount`...it seems to be the app saying to the implementation: "I won't take less than 2...but if you give me more...I'm cool with that"...rather than a more precise "just give me 2"...which would imply a more exact number...possibly named something like `imageCount`.  I assumed it was written that way because some implementations might be optimized for some larger number.

With that in mind...since the prominent use-case for Metal is triple-buffering...I took Metal to be one of those implementations that might just want to say "I know you can get by with 2...but 3 works better for me...so you're getting 3".

I'm happy to be convinced otherwise...and to drop `minImageCount` back to 2 if my assumptions are off-base.

Reply:
The [CAMetalLayer docs](https://developer.apple.com/documentation/quartzcore/cametallayer/2938720-maximumdrawablecount) suggest only 2 or 3 is possible for Metal. I guess if the user requests 2, then set `maximumDrawableCount` to 2, and to 3 for 3.  

`pSurfaceCapabilities->minImageCount` should definitely stay at 2 as that is a true representation of the underlying Metal capability. There are definitely circumstances (latency-critical apps) where 2 (ie double-buffering) would be preferred. 

Technically according to the spec, maximumDrawableCount could be 3 if the user requests mimimum 2 as you suggest, but that's unlikely to be what a user would expect.

Reply:
Does this change forget to also set maxImageCount?  Would that be left undefined?